src/handlers/issue-deduplication.ts

```diff
@@ -30,52 +77,82 @@ export async function issueChecker(context: Context<"issues.opened" | "issues.ed
+    });
+    return;
+  }
+
+  // Use the latest issue data
+  let issueBody = originalIssue.body;
   if (!issueBody) {
-    logger.info("Issue body is empty", { issue });
+    logger.info("Issue body is empty", { originalIssue });
     return;
   }
   issueBody = removeFootnotes(issueBody);
   const similarIssues = await supabase.issue.findSimilarIssues({
-    markdown: issue.title + removeFootnotes(issueBody),
-    currentId: issue.node_id,
-    threshold: context.config.warningThreshold,
+    markdown: originalIssue.title + removeFootnotes(issueBody),
+    currentId: originalIssue.node_id,
+    threshold: context.config.dedupeWarningThreshold,
   });
   if (similarIssues && similarIssues.length > 0) {
     let processedIssues = await processSimilarIssues(similarIssues, context, issueBody);
     processedIssues = processedIssues.filter((issue) =>
       matchRepoOrgToSimilarIssueRepoOrg(payload.repository.owner.login, issue.node.repository.owner.login, payload.repository.name, issue.node.repository.name)
     );
-    const matchIssues = processedIssues.filter((issue) => parseFloat(issue.similarity) / 100 >= context.config.matchThreshold);
+    const matchIssues = processedIssues.filter((issue) => parseFloat(issue.similarity) / 100 >= context.config.dedupeMatchThreshold);
     if (matchIssues.length > 0) {
-      logger.info(`Similar issue which matches more than ${context.config.matchThreshold} already exists`, { matchIssues });
+      logger.info(`Similar issue which matches more than ${context.config.dedupeMatchThreshold} already exists`, { matchIssues });
       //To the issue body, add a footnote with the link to the similar issue
       const updatedBody = await handleMatchIssuesComment(context, payload, issueBody, processedIssues);
       issueBody = updatedBody || issueBody;
       await octokit.rest.issues.update({
         owner: payload.repository.owner.login,
         repo: payload.repository.name,
-        issue_number: issue.number,
+        issue_number: originalIssue.number,
         body: issueBody,
         state: "closed",
         state_reason: "not_planned",
       });
       return;
     }
     if (processedIssues.length > 0) {
-      logger.info(`Similar issue which matches more than ${context.config.warningThreshold} already exists`, { processedIssues });
-      await handleSimilarIssuesComment(context, payload, issueBody, issue.number, processedIssues);
+      logger.info(`Similar issue which matches more than ${context.config.dedupeWarningThreshold} already exists`, { processedIssues });
+      await handleSimilarIssuesComment(context, payload, issueBody, originalIssue.number, processedIssues);
       return;
     }
   } else {
     //Use the IssueBody (Without footnotes) to update the issue when no similar issues are found
     //Only if the issue has "possible duplicate" footnotes, update the issue
-    if (checkIfDuplicateFootNoteExists(issue.body || "")) {
+    if (checkIfDuplicateFootNoteExists(issueBody || "")) {
```

Comment:
I'll just footnote logic makes me think that we should pre-fix the footnote IDs with something more sensible and precise.

Prefixing with a zero seems random.

Prefixing with something like `deduplication-` makes more sense to me